fix: minor documentation and test improvements

sangelovic opened the pull request:
This PR:

* removes unnecessary `releaseName()` calls in integration tests,
* adds a custom gtest message if a connection test fails, pointing the user to the potential cause of the issue,
* adds more documentation on how to test sdbus-c++ build.

Fixes #162 
Fixes #155

riuriuriu commented on docs/using-sdbus-c++.md:
````diff
@@ -100,6 +100,19 @@ Tip: If you get `ERROR: Program or command 'getent' not found or not executable`
+
+```
+$ cd build
+$ cmake .. -DBUILD_TESTS=ON
+$ make
+$ sudo cp ../tests/integrationtests/files/org.sdbuscpp.integrationtests.conf /etc/dbus-1/system.d/
+$ make test
````
Same. Examples using make assume the Makefile CMake Generator. It's likely that someone uses another generator, like ninja. I'd suggest to always invoke build and test via CMake which knows the generator.
```suggestion
$ cmake .. -DBUILD_TESTS=ON
$ sudo cp ../tests/integrationtests/files/org.sdbuscpp.integrationtests.conf /etc/dbus-1/system.d/
$ cmake --build . --target test
```

riuriuriu commented on tests/integrationtests/DBusConnectionTests.cpp:
```diff
@@ -54,8 +54,8 @@ TEST(Connection, CanRequestRegisteredDbusName)
-    ASSERT_NO_THROW(connection->requestName(INTERFACE_NAME));
-    connection->releaseName(INTERFACE_NAME);
+    ASSERT_NO_THROW(connection->requestName(INTERFACE_NAME))
+        << "Perhaps you've forgotten to copy `org.sdbuscpp.integrationtests.conf` file to `/etc/dbus-1/system.d` directory before running the tests?";
```
Very nice helper!

riuriuriu reviewed (approved):
Just added a 2 suggestions for build test commands. It might be applicable to the general build instructions too but that's not part of this MR.

sangelovic commented on README.md:
```diff
@@ -40,7 +40,7 @@ $ sudo make install
 * `BUILD_TESTS` [boolean]
 
-  Option for building sdbus-c++ unit and integration tests, invokable by `make test`. That incorporates downloading and building static libraries of Google Test. Default value: `OFF`. Use `-DBUILD_TESTS=ON` to enable building the tests. With this option turned on, you may also enable/disable the following options:
+  Option for building sdbus-c++ unit and integration tests, invokable by `make test` (Note: before invoking `make test`, make sure you copy `tests/integrationtests/files/org.sdbuscpp.integrationtests.conf` file to `/etc/dbus-1/system.d` directory). That incorporates downloading and building static libraries of Google Test. Default value: `OFF`. Use `-DBUILD_TESTS=ON` to enable building the tests. ...
```
Good idea, I agree.

sangelovic commented:
@riuriuriu Since this MR is also about documentation updates, I've incorporated your suggestion on make vs cmake in build examples, and I've updated it in other remaining places as well. Thanks.
